src/basic/fd-util.c

```diff
@@ -712,3 +708,13 @@ int btrfs_defrag_fd(int fd) {
 
         return 0;
 }
+
+char *format_proc_fd_path(char buf[static PROC_FD_PATH_MAX], int fd) {
+        assert(buf);
+
+        if (fd < 0)
+                return NULL;
+
+        assert_se(snprintf(buf, PROC_FD_PATH_MAX, "/proc/self/fd/%i", fd) < (int) PROC_FD_PATH_MAX);
```

Comment:
Maybe, `snprintf_ok()`?

Reply:
to be frank I think snprintf_ok() could use more love. The problem is that it casts the return value of snprintf() immediately to size_t even though the call might theoretically return -1 which it then would convert to INT_MAX i figure, and weird things happen. hence I figure I'd write the test here manually...

but you are right, we should fix snprintf_ok() instead to be entirely robust. will do.